src/Math-Numerical/Integer.extension.st

```diff
@@ -1,5 +1,54 @@
 Extension { #name : #Integer }
 
+{ #category : #'*Math-Numerical' }
+Integer >> findNK [
+	" Reverse binomial coefficient. Answer a <Collection> with all n and k such that n take: k = self. Elements in the answered Collection should be read as paired. Each pair represents (n,k) in the binomial coefficient formula. 
+	See https://math.stackexchange.com/a/103385/205 for details. "
+
+	| k |
+	[ self > 1 ] assert.
+    k := 0.
+	^ Array streamContents: [ : stream | 
+	    [ true ] whileTrue: [ 
+			| nmin nmax choose |
+			k := k + 1.
+			((2 * k + 1) * self) <= (4 ** k)
+				ifTrue: [ ^ stream contents ].
+			nmin := k firstOver: (k factorial * self).
+			nmax := nmin + k + 1.
+			nmin := nmin max: (2 * k).
+	      choose := nmin asInteger take: k.
+			nmin to: nmax do: [ : n |
+				choose = self
+					ifTrue: [  
+						stream nextPutAll: { n asInteger . k asInteger }.
+						k < (n - k)
+							ifTrue: [ stream nextPutAll: { n asInteger . (n - k) asInteger } ] ].
+	            choose := choose * (n + 1).
+	            choose := (choose / (n + 1 - k)) ceiling ] ] ]
+
+]
+
+{ #category : #'*Math-Numerical' }
+Integer >> firstOver: c [
```

Comment:
firstOver: is not really intention revealing...

If I understand, that's solving `n^self>=c`, that is `n=(c^(1/self)) ceiling`, or in other words, that's `(c nthRoot: self) ceiling`.
Both Cuis and Squeak have `nthRoot:`. If Pharo lacks such method, then it shall be added to Pharo.

Reply:
I quickly checked but `2 nthRoot: 3`in Squeak has different output than `2 firstOver: 3`.